src-electron/generator/helper-zcl.js

```diff
@@ -144,21 +144,36 @@ async function zcl_structs(options) {
     )
   }
   structs = await zclUtil.sortStructsByDependency(structs)
-  structs.forEach((st) => {
+  for (const st of structs) {
     st.struct_contains_array = false
     st.struct_has_fabric_sensitive_fields = false
     st.has_no_clusters = st.struct_cluster_count < 1
     st.has_one_cluster = st.struct_cluster_count == 1
     st.has_more_than_one_cluster = st.struct_cluster_count > 1
-    st.items.forEach((i) => {
-      if (i.isArray) {
-        st.struct_contains_array = true
-      }
-      if (i.isFabricSensitive) {
-        st.struct_has_fabric_sensitive_fields = true
+    const checkTransitive = async (items) => {
+      for (const i of items) {
+        if (i.isArray) {
+          st.struct_contains_array = true
+        }
+        if (i.isFabricSensitive) {
+          st.struct_has_fabric_sensitive_fields = true
+        }
+        if (
+          !st.struct_contains_array ||
+          !st.struct_has_fabric_sensitive_fields
+        ) {
+          let sis = await queryZcl.selectAllStructItemsByStructName(
```

Comment:
Struct name is not unique; you need to know which cluster is involved too, no?

Furthermore, the same name could be a struct or some other type too (especially across clusters).

Reply:
Ouch, thanks for catching that, this was copied from the `checkForDoubleNestedArray` clause below, which apparently has this bug

Do you know if there's already a utility which selects the best match for a struct? i.e. within same cluster if it exists, global otherwise? If there isn't one, I'll write one up

Reply:
selectStructByNameAndClusterId

Reply:
`selectAllStructItemsById` did the trick, but unfortunately there was no way to get the ID of the struct from the `STRUCT_ITEM`, so I had to do some deep surgery to expose that field 🤣

But looks a lot better, thanks for the tip!